Mathlib/RingTheory/PowerSeries/Schroder.lean

```diff
@@ -0,0 +1,199 @@
+/-
+Copyright (c) 2025 Weijie Jiang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Weijie Jiang
+-/
+module
+
+public import Mathlib.Combinatorics.Enumerative.Schroder
+public import Mathlib.RingTheory.PowerSeries.Basic
+
+/-!
+# Schröder Numbers Power Series
+
+This file defines lemmas and theorems about the power series for large and small Schröder numbers.
+
+## Main Definitions
+* `PowerSeries.largeSchroderPowerSeries`: The power series for large Schröder numbers.
+* `PowerSeries.smallSchroderPowerSeries`: The power series for small Schröder
+
+## Main Results
+* `PowerSeries.largeSchroder_eq_one_add_X_mul_largeSchroder_add_X_mul_largeSchroder_pow`:
+  The functional equation for the large Schröder numbers power series.
+
+## TODO
+
+* Prove the functional equation for the small Schröder numbers power series.
+
+-/
+
+@[expose] public section
+
+open Finset BigOperators Nat
+
+namespace PowerSeries
+
+/-- The power series for large Schröder numbers -/
+def largeSchroderPowerSeries : PowerSeries ℕ :=
+  PowerSeries.mk largeSchroder
+
+@[simp]
+lemma largeSchroderPowerSeries_coeff (n : ℕ) :
+  (coeff n) largeSchroderPowerSeries = largeSchroder n := by
+  simp [largeSchroderPowerSeries]
+
+@[simp]
+lemma largeSchroderPowerSeries_constantCoeff :
+  constantCoeff largeSchroderPowerSeries = 1 := by
+  rw [← PowerSeries.coeff_zero_eq_constantCoeff_apply]
+  simp [largeSchroderPowerSeries_coeff, largeSchroder_zero]
+
+@[simp]
+lemma largeSchroder_sum_coeff (n : ℕ) (hn : 0 < n) :
```

Comment:
```suggestion
lemma sum_coeff_X_mul_largeSchroder (n : ℕ) (hn : 0 < n) :
```
Also this is a weird lemma. This still holds if you replace `largeSchroder` with any function (= the lemma is not general enough)
Unfortunately I don't immediately have suggestion on what the right statement should be.